adal/src/main/java/com/microsoft/aad/adal/BrokerAccountServiceHandler.java

```diff
@@ -204,11 +204,35 @@ public void onError(Throwable throwable) {
         }
 
         final Throwable throwable = exception.getAndSet(null);
+        //AuthenticationException with error code BROKER_AUTHENTICATOR_NOT_RESPONDING will be thrown if there is any exception thrown during binding the service.
         if (throwable != null) {
-            Logger.e(TAG, "Didn't receive the activity to launch from broker: " + throwable.getMessage(), "", null, throwable);
+            if (throwable instanceof RemoteException) {
+                Logger.e(TAG, "Get error when trying to get token from broker. ",
+                        throwable.getMessage(), ADALError.BROKER_AUTHENTICATOR_NOT_RESPONDING, throwable);
+                throw new AuthenticationException(ADALError.BROKER_AUTHENTICATOR_NOT_RESPONDING,
+                        throwable.getMessage(),
+                        throwable);
+            } else if (throwable instanceof InterruptedException) {
+                Logger.e(TAG, "The broker account service binding call is interrupted. ",
+                        throwable.getMessage(), ADALError.BROKER_AUTHENTICATOR_EXCEPTION, throwable);
+                throw new AuthenticationException(ADALError.BROKER_AUTHENTICATOR_NOT_RESPONDING,
+                        throwable.getMessage(),
+                        throwable);
+            } else {
+                Logger.e(TAG, "Didn't receive the activity to launch from broker. ",
+                        throwable.getMessage(), ADALError.BROKER_AUTHENTICATOR_NOT_RESPONDING, throwable);
+                throw new AuthenticationException(ADALError.BROKER_AUTHENTICATOR_NOT_RESPONDING,
+                        "Didn't receive the activity to launch from broker: " + throwable.getMessage(),
+                        throwable);
+            }
         }
 
-        return bundleResult.getAndSet(null);
+        if (bundleResult.get() == null) {
+            Logger.e(TAG, "Get null intent from broker.", "", ADALError.BROKER_AUTHENTICATOR_NOT_RESPONDING);
```

Comment:
I don't think this is required.

Reply:
This is a public method. I am thinking if we remove the null check and move it to its calling method. Then should we need to let the developer know as well?